pandas/core/arrays/categorical.py

```diff
@@ -1764,18 +1757,10 @@ def ravel(self, order="C"):
         )
         return np.array(self)
 
-    def view(self):
-        """
-        Return a view of myself.
-
-        For internal compatibility with numpy arrays.
-
-        Returns
-        -------
-        view : Categorical
-           Returns `self`!
-        """
-        return self
+    def view(self, dtype=None):
+        if dtype is not None:
+            raise NotImplementedError(dtype)
+        return self._constructor(values=self._codes, dtype=self.dtype, fastpath=True)
```

Comment:
the default implementation does not work here? (or is this more efficient?)

Reply:
This is more efficient, yes (note the fastpath kwarg)

Reply:
Can you add a note for that? (eg "override base implementation to use fastpath")